proto/message.py

```diff
@@ -520,6 +541,29 @@ def __init__(
         # Create the internal protocol buffer.
         super().__setattr__("_pb", self._meta.pb(**params))
 
+    def __dir__(self):
+        desc = type(self).pb().DESCRIPTOR
+        names = {f_name for f_name in self._meta.fields.keys()}
+        names.update(m.name for m in desc.nested_types)
+        names.update(e.name for e in desc.enum_types)
+        names.update(dir(object()))
```

Comment:
What does this do?

Reply:
It's including all the fields message fields, nested message and enum types, and boilerplate methods and attributes inherited from `object`. The important one is fields, since they weren't visible before, but since this is customizing `__dir__` we need to explicitly add all the others back in too.